exchanges/account/account_types.go

```diff
@@ -26,6 +26,23 @@ type Service struct {
 	mu               sync.Mutex
 }
 
+func (s *Service) trackNewAccounts(exch string) (*Accounts, error) {
```

Comment:
As I said in the comment, I'm not a fan of this because the function requires a lock before the call. I find these kinds of functions bad form since they have prerequisites to avoid races that is not clear. 

Please add a comment to the function that a lock is required which helps alleviate it somewhat.

Reply:
Also please add unit test

Reply:
Another approach would be to replace the `Mutex` with an `RWMutex`, and in `trackNewAccounts`, we could check whether the lock has been acquired before the function is called.

```
func (s *Service) trackNewAccounts(exch string) (*Accounts, error) {
	if s.mu.TryLock() {
		s.mu.Unlock()
		return nil, errors.New("must lock before calling trackNewAccounts")
	}
```

Reply:
I appreciate the concept! It's only a 2ns check for the function if the lock is already in place. I am happy with a comment, but I also am happy with your suggestion

Reply:
Not bad; concurrent locks can spoof this though e.g. on startup, but mitigated by implementation test cases. 👍 4 me.

I think we can defer the `RWMutex` unless there are some real world bottlenecks that has been seen and it can be provably benchmarked, because an internal update is pretty quick, writes and reads are going to be similar in a websocket context and can possibly trip over each other.

Reply:
OK, Let's go ahead with just adding the comment for now.